snuba/datasets/cdc/cdcprocessors.py

```diff
@@ -1,12 +1,37 @@
+from abc import ABC, abstractclassmethod
+from typing import Any, Mapping, Type
+
 from snuba.processor import MessageProcessor
 
 KAFKA_ONLY_PARTITION = 0  # CDC only works with single partition topics. So partition must be 0
 
 
+class CDCMessageRow(ABC):
+    """
+    Takes care of the data transformation from WAL to clickhouse and from
+    bulk load to clickhouse. The goal is to keep all these transformation
+    function in the same place because they ultimately have to be consistent
+    with the Clickhouse schema.
+    """
+
+    @abstractclassmethod
+    def from_wal(cls, offset, columnnames, columnvalues):
+        raise NotImplementedError
+
+    @abstractclassmethod
+    def from_bulk(cls, row):
+        raise NotImplementedError
+
+    @abstractclassmethod
+    def to_clickhouse(self) -> Mapping[str, Any]:
```

Comment:
This should probably return `snuba.writer.TableRow` at this point?

Reply:
true